frontend/pages/ManageControlsPage/MacOSSettings/cards/DiskEncryption/DiskEncryption.tsx

```diff
@@ -20,18 +22,35 @@ interface IDiskEncryptionProps {
 }
 
 const DiskEncryption = ({ currentTeamId }: IDiskEncryptionProps) => {
-  const { isPremiumTier, config } = useContext(AppContext);
+  const { isPremiumTier, config, setConfig } = useContext(AppContext);
   const { renderFlash } = useContext(NotificationContext);
 
   const defaultShowDiskEncryption = currentTeamId
     ? false
     : config?.mdm.macos_settings.enable_disk_encryption ?? false;
 
+  const [isLoadingTeam, setIsLoadingTeam] = useState(true);
+
   const [showAggregate, setShowAggregate] = useState(defaultShowDiskEncryption);
   const [diskEncryptionEnabled, setDiskEncryptionEnabled] = useState(
     defaultShowDiskEncryption
   );
 
+  // because we pull the default state for no teams from the config,
+  // we need to update the config when the user toggles the checkbox
```

Comment:
As comment states, any time the "No team" settings are updated, we need to re-pull the app config and set it in state.